event-store-plans.md

```diff
@@ -0,0 +1,174 @@
+# Event Store Improvements for V7
+
+This document is brainstorming for improvements to the Event Store functionality in Marten V7. This does not include features meant for
+"CritterStackPro" products, but does include some necessary baseline infrastructure for planned "CritterStackPro" features.
+
+In some cases, this document outlines "poor man's" subset strategies for features that we will build in a more robust (and inevitably more complicated) way 
+in "CritterStackPro"
+
+The goal is to start a discussion about what is and isn't going to be in scope for V7. When this document is "complete", the output is a new set
+of GitHub issues marked for the V7 release. 
+
+
+## Raise events from asynchronous projection updates
+
+I'm going to allow Oskar address this one:-)
+
+From my side, I think the `ISessionListener` approach we have today is weak and should be a little more "outbox'd" such that it only fires off when
+the transaction is complete. 
+
+For the Wolverine integration, I'd like to be able to enroll in Wolverine's outbox as part of processing a page of events in the daemon to send out messages using the 
+aggregation data as part of the outgoing message.
+
+More generically, we need a good way to emit new events that will be processed as appends within the same daemon transaction. This might be a little tricky and will
+require a spike.
+
+**Gotta do this in a way such that users can opt into whether the new events should be emitted during projection rebuilds**
+
+
+## Sharding the Event Store tables
```

Comment:
I think that we should also give the possibility to scan per tenant and stream type. This could speed up a lot of processing, but of course, it would mean that we don't have global ordering between partitions. Still, in many cases this should be good enough to speed up performance and allow easier database size management.